easybuild/easyconfigs/t/TomoPy/TomoPy-1.11.0-foss-2020b.eb

```diff
@@ -0,0 +1,53 @@
+# Authors: Joachim Hein, Zdenek Matej, Lund University
+
+easyblock = 'PythonPackage'
+
+name = 'TomoPy'
+version = '1.11.0'
+
+github_account = 'tomopy'
+homepage = "https://tomopy.readthedocs.io/en/latest/"
+description = """
+Tomographic reconstruction creates three-dimensional views of an object by 
+combining two-dimensional images taken from multiple directions, for example, 
+this is how a CAT (computer-aided tomography) scanner generates 3D 
+views of the heart or brain.
+"""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('SciPy-bundle', '2020.11'),
+    ('scikit-image', '0.18.1'),
+    ('OpenCV', '4.5.1', '-contrib'),
+    ('imkl', '2020.4.304'),
+    ('numexpr', '2.8.1'),
+    ('DXchange', '0.1.8'),
+    ('PyWavelets', '1.1.1'),
+]
+
+builddependencies = [
+    ('scikit-build', '0.11.1'),
+    ('CMake', '3.18.4'),
+    ('Ninja', '1.10.1'),
+]
+
+use_pip = True
+download_dep_fail = True
+sanity_pip_check = True
+
+sources = [{
+    'filename': '%(name)s-%(version)s.tar.gz',
+    'git_config': {
+        'url': 'https://github.com/tomopy',
+        'tag': '%(version)s',
+        'recursive': True,
```

Comment:
I would prefer to see an external PTL, with the benefit that it would actually be useful elsewhere, it would have a tagged release, it would remove the need to do a git checkout here so we can actually have checksums.

Seems quote easily doable with TOMOPY_EXTERNAL_LIBRARIES and TOMOPY_EXTERNAL_INCLUDE_DIRS.

Reply:
The only reason i see for using the git checkout is because it uses a git submodule for the PTL dependency, which, like many other software projects do, they incorrectly "bundle" in their sources. 
Just like we do with lots of other software, we configure/patch the softwares to use this dependency externally instead. I.e. 1. make another easyconfig for PTL
2. switch to sources without PTL "bundled" (thus gaining the benefits of using a proper release than git checkout, meaning we can now also use checksums) 
3. configure/patch TomoPy to use it's PTL dependency externally instead.

Reply:
@jhein32 I guess the PTL dependency is in [tomopy/source](https://github.com/tomopy/tomopy/tree/1.11.0/source) directory and it is this project: [jrmadsen/PTL](https://github.com/jrmadsen/PTL). I guess this we can try to fix. That is really a serious bug.